controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/LoadSSTTask.java

```diff
@@ -0,0 +1,230 @@
+import com.pinterest.rocksplicator.controller.bean.ClusterBean;
+import com.pinterest.rocksplicator.controller.bean.HostBean;
+import com.pinterest.rocksplicator.controller.bean.Role;
+import com.pinterest.rocksplicator.controller.bean.SegmentBean;
+import com.pinterest.rocksplicator.controller.bean.ShardBean;
+import com.pinterest.rocksplicator.controller.util.AdminClientFactory;
+import com.pinterest.rocksplicator.controller.util.ShardUtil;
+import com.pinterest.rocksplicator.controller.util.ZKUtil;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import javax.inject.Inject;
+
+/**
+ * This task loads sst files from S3 to a cluster managed by the controller.
+ *
+ * @author Ang Xu (angxu@pinterest.com)
+ */
+public class LoadSSTTask extends TaskBase<LoadSSTTask.Param> {
+  private static final Logger LOG = LoggerFactory.getLogger(LoadSSTTask.class);
+  @Inject
+  private CuratorFramework zkClient;
+
+  @Inject
+  private AdminClientFactory clientFactory;
+
+  public LoadSSTTask(String segment,
+                     String s3Bucket,
+                     String s3Prefix,
+                     int concurrency,
+                     int rateLimitMbs) {
+    this(
+        new Param().setSegment(segment)
+            .setS3Bucket(s3Bucket)
+            .setS3Prefix(s3Prefix)
+            .setConcurrency(concurrency)
+            .setRateLimitMbs(rateLimitMbs)
+    );
+  }
+
+  public LoadSSTTask(Param param) {
+    super(param);
+  }
+
+  @Override
+  public void process(Context ctx) throws Exception {
+    final String clusterName = ctx.getCluster();
+    final String segment = getParameter().getSegment();
+
+    ClusterBean clusterBean = ZKUtil.getClusterConfig(zkClient, clusterName);
+    if (clusterBean == null) {
+      LOG.error("Failed to get config for cluster {}.", clusterName);
+      ctx.getTaskQueue().failTask(ctx.getId(), "Failed to read cluster config from zookeeper.");
+      return;
+    }
+    SegmentBean segmentBean = clusterBean.getSegments()
+        .stream()
+        .filter(s -> s.getName().equals(segment))
+        .findAny()
+        .orElse(null);
+    if (segmentBean == null) {
+      String errMsg = String.format("Segment %s not in cluster %s.", segment, clusterName);
+      LOG.error(errMsg);
+      ctx.getTaskQueue().failTask(ctx.getId(), errMsg);
+      return;
+    }
+
+    final ExecutorService executor = Executors.newFixedThreadPool(getParameter().getConcurrency());
+
+    // first pass load sst to masters
+    List<Future<Boolean>> futures = new ArrayList<>(segmentBean.getHosts().size());
```

Comment:
first pass and second pass code looks pretty similar, put them into one function?